app/models/docker_registry.rb

```diff
@@ -37,4 +38,16 @@ def prefixed_url(image_name)
   def self.humanize_class_name(_name = nil)
     _("Docker/Registry")
   end
+
+  private
+
+  def attempt_login
+    credentials = { 'username' => username, 'password' => password }
+    connection = ::Docker::Connection.new(
+      url,
+      credentials)
+    ::Docker.authenticate!(credentials, connection)
+  rescue => e
```

Comment:
I can but does it deserve special treatment? I left it to catch all StandardError as to catch network problems etc...